modules/core/shared/src/main/scala/lucuma/core/optics/README.md

```diff
@@ -10,6 +10,9 @@ When we compose a split mono and a split epi end-to-end in either direction we e
 
 A `Format` is a weaker `Prism` where a *subset* of `A` forms a split epi with `B`. Every `Prism` is a `Format` where the split epi happens to be an `Iso`; and every `SplitEpi` forms a `Prism` where the subset of `A` is `A` itself.
 
+A `ValidSplitEpi` is a reformulation of a `Format`, where elements of `A` that don't map to an element of `B` return an element of `E` instead. A `Format[A, B]` is a `ValidSplitEpi[Unit, A, B]`.
+
+A `ValidWedge` is to a `Wedge` what a `ValidSplitEpi` is to a `SplitEpi`. In other words, it allows for reverse normalization.
```

Comment:
I like this `ValidSplitEpi` concept but I wonder about the "split epi"ness of it.  As I understand them, split epi and wedge are special `Iso`s where you can always go back and forth between the `A` and the `B` even if a cycle doesn't yield the same value you started with.  As you rightly point out `ValidSplitEpi` is a reformulation of `Format`, a `ValidFormat` if you will.  I guess what I'm asking is why not drop the `ValidWedge` and call the `ValidSplitEpi` a `ValidFormat`?

Reply:
That's exactly what we had before (`ValidSplitEpi` named as `ValidFormat`). This had similar laws than `Format`.

However this fell short for us in some cases, specifically when we truncating decimal numbers, since in that case we are normalizing both ways, and that's how `ValidWedge` was born (and `ValidFormat` renamed to `ValidSplitEpi`).

I fiddled a bit with the names and settled on these which seem to lessen the mental burden of introducing new concepts by expanding on existing ones. The `Valid*` versions of `SplitEpi` and `Wedge` are basically the same as those for the subset of valid instances of `A`.

Also, I'm reintroducing `ValidFormat` in the next PR as a generalization of `ValidSplitEpi` and `ValidWedge` since they share the same `getValid`/`reverseGet` methods and it makes it more convenient for APIs to receive either.

All that said, I'm completely open to renaming.

Reply:
:+1: I see, that makes sense and I have no better ideas.

Reply:
There is quite a long history of discussion of this issue. In the distant past, we ran into issues with `ValidFormat` when we realized we needed to be able to truncate decimals in the UI. We discussed a couple of alternatives: 

1. Weaken the laws to essentially be those of the new `ValidWedge`.
2. Create specific types for things like `TruncatedBigDecimal`, `TruncatedRefinedBigDecimal`, `TruncatedRA`, etc. that guaranteed only `n` decimal places so the `ValidFormat` laws would pass.

We opted for number 2, but reassessed it recently. We decided to have 2 versions with different guarantees instead.